tools/cli/helpers/docker-config.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * External dependencies
+ */
+import { readFileSync, writeFileSync, existsSync } from 'fs';
+import yaml from 'js-yaml';
+import path from 'path';
+import { merge } from 'lodash';
+
+export const dockerFolder = `tools/docker`;
+const overrideConfigFile = '.docker-env.json';
+
+const mergeDockerVolumeMappings = ( mainMapping, overrideMapping ) => {
+	const volumesMapping = overrideMapping.slice(); // make a copy of an array;
+
+	const typeLocPaths = overrideMapping.map( map => map.split( ':' )[ 0 ] );
+	const defaultLocPaths = mainMapping.map( map => map.split( ':' )[ 0 ] );
+	defaultLocPaths.forEach( ( defaultPath, id ) => {
+		if ( ! typeLocPaths.includes( defaultPath ) ) {
+			volumesMapping.unshift( mainMapping[ id ] );
+		}
+	} );
+
+	return volumesMapping;
+};
+
+/**
+ * DEPRECATED. Parses compose-volumes.yml and adds it's contents into docker config
+ *
+ * @param {object} config - Docker configuration
+ * @returns {object} config
+ */
+const getDeprecatedVolumesMapping = config => {
+	const volumesFile = `${ dockerFolder }/compose-volumes.yml`;
+	const volumes = yaml.load( readFileSync( volumesFile, 'utf8' ) );
+
+	// convert array of docker volumes into a object of local/docker paths
+	const volumesObj = volumes.reduce( ( acc, volume ) => {
+		const [ localPath, dockerPath ] = volume.split( ':' );
+		let relPath = path.relative( '.', 'tools/docker/' + localPath );
+		if ( ! relPath ) {
+			relPath = '.';
+		}
+		acc[ relPath ] = dockerPath;
+		return acc;
+	}, {} );
+
+	config.dev = {};
+	config.dev.mappings = volumesObj;
+	return config;
+};
+
+/**
+ * DEPRECATED. Parses compose-extras.yml and adds it's contents into docker config
+ *
+ * @param {object} config - Docker configuration
+ * @returns {object} config
+ */
+const getDeprecatedExtras = config => {
+	const extrasFile = `${ dockerFolder }/compose-extras.yml`;
+	const extras = yaml.load( readFileSync( extrasFile, 'utf8' ) );
+	delete extras.version;
+	config.dev.extras = extras;
+	return config;
```

Comment:
As above, you're assuming you can stomp on whatever was in there previously. And you're assuming that something set `config.dev` earlier.
```suggestion
	return merge( config, { dev: { extras: extras } } );
```